dpnp/backend/extensions/lapack/gesv.cpp

```diff
@@ -42,49 +43,150 @@ namespace mkl_lapack = oneapi::mkl::lapack;
 namespace py = pybind11;
 namespace type_utils = dpctl::tensor::type_utils;
 
-typedef sycl::event (*gesv_impl_fn_ptr_t)(sycl::queue,
+void common_gesv_checks(sycl::queue &exec_q,
+                        dpctl::tensor::usm_ndarray coeff_matrix,
+                        dpctl::tensor::usm_ndarray dependent_vals,
+                        const py::ssize_t *coeff_matrix_shape,
+                        const py::ssize_t *dependent_vals_shape,
+                        const int expected_coeff_matrix_ndim,
+                        const int min_dependent_vals_ndim,
+                        const int max_dependent_vals_ndim)
+{
+    const int coeff_matrix_nd = coeff_matrix.get_ndim();
+    const int dependent_vals_nd = dependent_vals.get_ndim();
+
+    if (coeff_matrix_nd != expected_coeff_matrix_ndim) {
+        throw py::value_error("The coefficient matrix has ndim=" +
+                              std::to_string(coeff_matrix_nd) + ", but a " +
+                              std::to_string(expected_coeff_matrix_ndim) +
+                              "-dimensional array is expected.");
+    }
+
+    if (dependent_vals_nd < min_dependent_vals_ndim ||
+        dependent_vals_nd > max_dependent_vals_ndim)
+    {
+        throw py::value_error("The dependent values array has ndim=" +
+                              std::to_string(dependent_vals_nd) + ", but a " +
+                              std::to_string(min_dependent_vals_ndim) +
+                              "-dimensional or a " +
+                              std::to_string(max_dependent_vals_ndim) +
+                              "-dimensional array is expected.");
+    }
+
+    // The coeff_matrix and dependent_vals arrays must be F-contiguous arrays
+    // for gesv
+    // with the shapes (n,n) and (n,nrhs) or (n,) respectively;
+    // for gesv_batch
+    // with the shapes (n,n,batch_size) and (n,nrhs,batch_size) or
+    // (n,batch_size) respectively
```

Comment:
I believe it would be more readable with the spaces
```suggestion
    // with the shapes (n, n) and (n, nrhs) or (n, ) respectively;
    // for gesv_batch
    // with the shapes (n, n, batch_size) and (n, nrhs,batch_size) or
    // (n, batch_size) respectively
```